cmd/operator/main.go

```diff
@@ -147,6 +144,10 @@ func main() {
 			logger.Fatalf("error when running daemon: %v", err)
 		}
 	}
+
+	http.HandleFunc("/rollout", makeRolloutHandler(logger, cfg))
```

Comment:
let's change the -http-addr flag a bit.

1. edit its help message, it looks odd

2. right now default is "", you can give:

```
defaultAddr == ""
if v:= os.Getenv("PORT"); v!="" {
  defaultAddr = ":8080"
}
```

Reply:
We actually need to check for more things here, since there are 3 scenarios (in order of precedence):

1. The `PORT` variable is in the environment
2. The `-http-addr` flag
3. The default ":8080"

Because there's a default and the env variable has precedence, I think we need to drop the checks for just one of `-cli` or `-http-addr`. So, by default, we can assume the developer is running the tool as a webserver listening on `:8080` unless you specify you want to run it as CLI (`-cli` flag).

Reply:
I understand the point of doing 2, but I think we can determine the port later in the program instead of passing a default to the flag based on a env variable. If we do 2, the user might become confuse when they run

```
./operator -cli ...
```
and get the error "only one of -cli or -http-addr can be used" just because `PORT` was set in their env variables.

Reply:
🤔 yeah I see. In that case, what if we only relied on `-cli` to determine cli mode or not?
alternatively we can introduce `-server` option, which is mutually exclusive with `-cli`, this seems cleaner.

Reply:
I renamed it to `port` expecting an `int`. We're getting the default from the environment while making the HTTP the default mode. So, `-cli` should be used to run it in the terminal.

Reply:
I don't recommend that. we still need to have a way of specifying addr e.g. `localhost:8080` vs `:8080` are different things. 

Also we should still use $PORT to use `:$PORT` if `-http-addr` is unspecified (e.g. on Cloud Run )

Reply:
Then, we can relied only `-cli` as you pointed out and make the HTTP server the default one with default addr `:8080` which can be overwritten by `$PORT`. However, if `-http-addr` is specified, that has the highest precedence.